qt/python/mantidqt/mantidqt/widgets/sliceviewer/models/model.py

```diff
@@ -535,6 +531,30 @@ def get_proj_matrix(self):
             return None
 
     # private api
+
+    @staticmethod
+    def _get_proj_matrix_from_basis_vectors(ws, proj_matrix):
+        ndims = ws.getNumDims()
+
+        # Array of True or False depending on if a dimension is in Q
+        qdims = np.array([ws.getDimension(i).getMDFrame().isQ() for i in range(ndims)])
```

Comment:
Minor point, but as the `qdims` is something constant about the workspace we should probably store this as an attribute as we use it again when determining the HKL and maybe have some private methods to e.g. `get_index_nonQ_dims` etc. - what do you think?